webapp/src/js/controllers/inbox.js

```diff
@@ -263,28 +263,24 @@ var feedback = require('../modules/feedback'),
       });
     });
 
-    $transitions.onStart({}, function(trans) {
-      if (trans.to().name.indexOf('reports') === -1 || trans.to().name.indexOf('contacts') === -1 || trans.to().name.indexOf('tasks') === -1 || trans.to().name.indexOf('messages.detail') === -1) {
-        $scope.unsetSelected();
+    $transitions.onBefore({}, (trans) => {
+      if (ctrl.enketoStatus.edited && ctrl.cancelCallback) {
+        $scope.navigationCancel({ to: trans.to(), params: trans.params() });
+        return false;
```

Comment:
`return false;` aborts the transition:
https://ui-router.github.io/guide/transitionhooks#aborting-a-transition